### ISSUE TITLE
Correctly describe the limitation of mounting subvolumes with different Btrfs mount options

### ISSUE THREAD

**Issue by jwillikers** [jwillikers/blog#12]
Btrfs-specific mount options are applied to all subvolumes from that Btrfs filesystem and cannot be altered.
This ability is planned but not yet available.
I believe I mention using such behavior in my blog posts, but this won't work.
I need to review all Btrfs posts and locate such disparities, correcting any bad instructions related to this behavior.

**Comment by martin-tk** [comment]
I'm planning my btrfs layout and came to the same realisation that mount options for disabling CoW and controlling compression algorithm and level don't work for subvolumes. More precisely, options used when mounting **first** subvolume of a given btrfs filesystem take effect for all other subvolumes for that particular filesystem.

The way I'm going to control those is by setting directory (or subvolume) attribute to disable CoW (e.g. `chattr +C /mnt/@/var/lib/containers` etc.).

To disable compression `chattr +m dir`. `chattr +c dir` for enabling compression, but the first subvolume mounted has to have `compress=zstd` mount option (looks like you can only use a single algorithm).

If you want to change compression level you can try `btrfs property set /mnt/subvolume compression zstd:7`.
Basically, you have to mount the first subvolume with `compress=zstd` and disable compression using `chattr` command per subvolume/directory basis.

For reference, first paragraph of man 5 btrfs explains that mount options for subvolumes may have no effect.
